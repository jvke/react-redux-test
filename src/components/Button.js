import React from 'react';
import styled from 'styled-components';
import { lighten, darken, adjustHue } from 'polished';

const Button = ({ children, ...otherProps }) => (
  <button {...otherProps}>{children}</button>
);

export default styled(Button)`
  &:focus {
    outline: none;
  }
  border: none;
  color: ${props => darken(0.1, props.theme.textColor)};
  border-radius: 4px;
  background: -webkit-gradient(linear, left top, right top, from(${props => lighten(0.3, adjustHue(50, props.theme.brand.light))}), to(${props => lighten(0.1, adjustHue(75, props.theme.brand.light))}));
  background: -moz-linear-gradient(linear, left top, right top, from(${props => lighten(0.3, adjustHue(50, props.theme.brand.light))}), to(${props => lighten(0.1, adjustHue(75, props.theme.brand.light))}));
  background: -o-linear-gradient(linear, left top, right top, from(${props => lighten(0.3, adjustHue(50, props.theme.brand.light))}), to(${props => lighten(0.1, adjustHue(75, props.theme.brand.light))}));
  padding: 5px 10px;
  &:hover {
    cursor: pointer;
  }
`;
