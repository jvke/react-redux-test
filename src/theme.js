export default {
  brand: {
    light: 'rgb(8, 62, 101)',
    dark: 'rgb(1, 20, 46)',
  },
  alert: {
    error: '#FF0000'
  },
  textColor: '#EEEEEE'
}
