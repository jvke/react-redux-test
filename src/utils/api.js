import { put, call } from 'redux-saga/effects';
const handleSagaAPICall = function* _handleSagaAPICall(fetchFn, successFn, failFn, endpoint, ...args) {
  try {
    // do axios call
    const results = yield call(fetchFn, endpoint, ...args);

    if (results.statusText === 'OK') {
      yield put(successFn(results.data))
    } else {
      throw new Error('Unknown error has occurred');
    }
  } catch (e) {
    if (e.response && e.response.data) {
      yield put(failFn(new Error(e.response.data.message)));
    } else {
      yield put(failFn(new Error(e.message)));
    }
  }
}

export { handleSagaAPICall };
