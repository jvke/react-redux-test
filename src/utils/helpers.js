import { ThemeProvider } from 'styled-components';
import React from 'react';
const wrapWithTheme = (theme) => {
  return (children) => (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  )
}

export { wrapWithTheme }
