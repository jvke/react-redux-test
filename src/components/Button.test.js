import React from 'react';
import ReactDOM from 'react-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Button from './Button';
import theme from '../theme';
import { wrapWithTheme } from '../utils/helpers';

import { ThemeProvider } from 'styled-components';

const wrapper = wrapWithTheme(theme);

describe('Button', () => {
  it('Button renders with no props specified', () => {
    const tree = renderer
      .create(
        <Button theme={theme}>MyButton</Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button renders with props specified', () => {
    const tree = renderer
      .create(<Button theme={theme} a={1} b={2} c={3}>MyButton</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
