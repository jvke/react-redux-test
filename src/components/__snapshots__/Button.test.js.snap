// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders with no props specified 1`] = `
.c0 {
  border: none;
  color: #d5d5d5;
  border-radius: 4px;
  background: -webkit-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  background: -moz-linear-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  background: -o-linear-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  padding: 5px 10px;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
>
  MyButton
</button>
`;

exports[`Button Button renders with props specified 1`] = `
.c0 {
  border: none;
  color: #d5d5d5;
  border-radius: 4px;
  background: -webkit-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  background: -moz-linear-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  background: -o-linear-gradient(linear,left top,right top,from(#5019ed),to(#670c94));
  padding: 5px 10px;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  cursor: pointer;
}

<button
  a={1}
  b={2}
  c={3}
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
>
  MyButton
</button>
`;
